--- ./PKRequest.inc.php.original
+++ ./PKRequest.inc.php.correto
@@ -282,7 +282,7 @@
 		$_this =& PKPRequest::_checkThis();
 
 		if (!isset($_this->_protocol)) {
-			$_this->_protocol = (!isset($_SERVER['HTTPS']) || strtolower($_SERVER['HTTPS']) != 'on') ? 'http' : 'https';
+			$_this->_protocol = ( isset($_SERVER['HTTPS']) || strtolower($_SERVER['HTTPS']) == 'on' || @$_SERVER["HTTP_X_FORWARDED_PROTO"] == 'https' ) ? 'https': 'http';
 			HookRegistry::call('Request::getProtocol', array(&$_this->_protocol));
 		}
 		return $_this->_protocol;
